settings.js

```diff
@@ -302,28 +302,35 @@ function findConflicts(schemas) {
 */
 var winprops = [];
 
-function winprop_match_p(meta_window, prop) {
-    let wm_class = meta_window.wm_class || "";
-    let title = meta_window.title;
-    if (prop.wm_class.constructor === RegExp) {
-        if (!wm_class.match(prop.wm_class))
-            return false;
-    } else if (prop.wm_class !== wm_class) {
-        return false;
-    }
-    if (prop.title) {
-        if (prop.title.constructor === RegExp) {
-            if (!title.match(prop.title))
+function _winprop_match_p_helper(user_attr_prop, _win_prop) {
+    // matches user-supplied conditions with null checking
+    if (user_attr_prop) {
+        let win_prop = String(_win_prop);
+        if (user_attr_prop.constructor === RegExp) {
+            // treat prop condition as regex
+            if (!win_prop.match(user_attr_prop))
                 return false;
```

Comment:
```suggestion
            return win_prop.match(user_attr_prop);
```
The logic here can be simplified. Also, I think the code here is self-explanatory enough.

Reply:
The whole thing could be condensed further, like,
```
return user_attr_prop.constructor === RegExp
    ? win_prop.match(user_attr_prop)
    : user_attr_prop === win_prop;
 ```
 but in my opinion this is only shorter, not simpler or more readable.